### DISCUSSION TITLE
how to make web api with mojo ?

### DISCUSSION THREAD

**Discussion by rectrick** [modular/modular#2389] [Q&A]
I am trying to build an application that relies heavily on artificial intelligence. I am trying to build an audio and video editor that relies on artificial intelligence through a web API, but there is nothing that shows that Mojo supports this. I was going to rely on Django Ninja because it provides data type, but I love it and I hope it does that. Is there How to produce api using mojo?

**Comment by rarebreed** [comment]
mojo is a very young language.  Not only are there no web frameworks like django, fastapi or flask available, but it also doesn't even have low-level socket libraries on which you could build a native http client/server.

At best, you could try to import python modules from mojo and create an app that way. However, it will be clumsy to build.  First, the mojo IDE integration won't understand the python types (so you may as well not be using type annotations).  That also means you won't get IDE completions since most things will be `PythonObject` types.  Also, while mojo can import python modules, it doesn't work the other way.  You can't load mojo packages into python.

You might be better off writing the backend in python, and executing a mojo binary in a child process to do the calculations and generate an output for the python web service to use.

**Comment by rectrick** [comment]
I tried asking ChatGPT and it gave me this answer. Can I actually take advantage of that?

https://chat.openai.com/share/701d0ad6-c5be-4860-88a9-1ce9520d9898
